datafusion/physical-plan/src/sorts/sort_preserving_merge.rs

```diff
@@ -105,6 +119,14 @@ impl SortPreservingMergeExec {
     }
 
     /// Sets the selection strategy of tied winners of the loser tree algorithm
+    ///
+    /// When true (the default) equal output rows are placed in the merged
+    /// stream when ready, which is faster but not stable (can vary from
+    /// run to run).
```

Comment:
From what I remember, round robin repartition choose the smaller index if other condition is equals (value and polled count is equal). If there are 3 streams, we choose 1, 2, 3 and back to 1, 2, 3 (because we count the polled count) until the value advanced. Do you know why the result vary from run to run and is not always the same?

Reply:
Poll may be pending so value is not yet ready?

Reply:
You are right -- I double checked the other parts of the code -- I think I misudnerstood the root need for this option.

I removed the discssion of polling when not ready.